File: base-backend-node/src/app/api/v1/routes/higs.ts
import { Hono } from 'hono';
import { getDatabases, DATABASE_ID } from '../../../services/appwriteClient';
import { Query } from 'node-appwrite';
import { Bindings } from '../../../core/types';

export const higsRouter = new Hono<{ Bindings: Bindings }>();

higsRouter.get('/events/latest', async (c) => {
  const db = getDatabases(c.env);
  try {
    const res = await db.listDocuments(DATABASE_ID, 'higs_events', [
      Query.equal('isActive', true),
      Query.orderDesc('$createdAt'),
      Query.limit(1)
    ]);
    if (res.documents.length) {
      return c.json({
        date: res.documents[0].date,
        registrationUrl: res.documents[0].registrationUrl
      });
    }
    return c.json({
      date: 'TBA',
      registrationUrl: 'https://luma.com/oyyhm42g'
    });
  } catch (e) {
    console.error(`Error fetching HIGS event:`, e);
    return c.json({
      date: 'TBA',
      registrationUrl: 'https://luma.com/oyyhm42g'
    });
  }
});

File: base-backend/app/api/v1/routes/higs.py
from fastapi import APIRouter
from app.services.appwrite_client import get_databases, DATABASE_ID
from appwrite.query import Query

router = APIRouter()

@router.get("/")
def get_active_higs_event():
    db = get_databases()
    try:
        response = db.list_documents(
            DATABASE_ID,
            "higs_events",
            [
                Query.equal("isActive", True),
                Query.order_desc("$createdAt"),
                Query.limit(1)
            ]
        )
        docs = response.get("documents", [])
        if docs:
            return {
                "date": docs[0].get("date"),
                "registrationUrl": docs[0].get("registrationUrl")
            }
        return {
            "date": "TBA",
            "registrationUrl": "https://luma.com/oyyhm42g"
        }
    except Exception as e:
        print(f"Error fetching HIGS event: {e}")
        return {
            "date": "TBA",
            "registrationUrl": "https://luma.com/oyyhm42g"
        }
